File: src/main/java/xyz/nkomarn/composter/network/protocol/packet/s2c/ClientboundRemoveEntityPacket.java
package xyz.nkomarn.composter.network.protocol.packet.s2c;

import io.netty.buffer.ByteBuf;
import io.netty.buffer.Unpooled;
import xyz.nkomarn.composter.entity.Entity;
import xyz.nkomarn.composter.network.protocol.Packet;

public class ClientboundRemoveEntityPacket extends Packet<ClientboundRemoveEntityPacket> {

    private int entityId;

    public ClientboundRemoveEntityPacket() {
    }

    public ClientboundRemoveEntityPacket(Entity entity) {
        this.entityId = entity.getId();
    }

    @Override
    public int getId() {
        return 0x1D;
    }

    @Override
    public ByteBuf encode() {
        return Unpooled.buffer().writeInt(entityId);
    }
}

File: src/main/kotlin/kyta/composter/protocol/packet/play/ClientboundRemoveEntityPacket.kt
package kyta.composter.protocol.packet.play

import kyta.composter.protocol.Packet
import kyta.composter.protocol.PacketSerializer
import kyta.composter.protocol.WriteBuffer
import xyz.nkomarn.composter.entity.Entity

data class ClientboundRemoveEntityPacket(val id: Int) : Packet {
    constructor(entity: Entity) : this(entity.id)

    companion object : PacketSerializer<ClientboundRemoveEntityPacket> {
        override fun serialize(packet: ClientboundRemoveEntityPacket, buffer: WriteBuffer) {
            buffer.writeInt(packet.id)
        }
    }
}